hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/mssql/MSSQLPredicatePushDownJdbcSqlConnectorTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.jdbc.mssql;
+
+import com.hazelcast.jet.sql.impl.connector.jdbc.PredicatePushDownJdbcSqlConnectorTest;
+import com.hazelcast.test.annotation.NightlyTest;
+import com.hazelcast.test.jdbc.MSSQLDatabaseProvider;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.experimental.categories.Category;
+
+import java.sql.SQLException;
+
+@Category(NightlyTest.class)
+public class MSSQLPredicatePushDownJdbcSqlConnectorTest extends PredicatePushDownJdbcSqlConnectorTest {
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        initializePredicatePushDownTestMSSQL(new MSSQLDatabaseProvider());
+    }
+
+    @Before
+    public void setUpForMSSQL() throws Exception {
+        //These statements have syntax incompatible with MSSQL
+        if (query.equals("SELECT name FROM people WHERE a AND b")) {
+            query = "SELECT name FROM people WHERE a = '1' AND b = '1'";
+        } else if (query.equals("SELECT name FROM people WHERE a OR b")) {
+            query = "SELECT name FROM people WHERE a = '1' OR b = '1'";
+        } else if (query.equals("SELECT name FROM people WHERE NOT c")) {
+            query = "SELECT name FROM people WHERE c != '1'";
+        } else if (query.equals("SELECT name FROM people WHERE a IS TRUE")) {
```

Comment:
can you please try to change MSSQLDialect to translate "invalid" SQL into MS-SQL-compatible one?

Reply:
We decided not to support BOOLEAN for MS SQL, because the BIT data type behaves there more like a numeric data type with 0, 1 as possible values.

Thes test was rewritten to use INT instead of BOOLEAN columns.